hpcbench/driver.py

```diff
@@ -697,45 +698,76 @@ def __init__(self, parent):
-            print("#!/bin/bash", file=ostr)
-            for command in commands:
-                if isinstance(command, list):
-                    print(' '.join(command), file=ostr)
-                else:
-                    print(command, file=ostr)
+            self._write_executor_script(ostr)
+        os.chmod(path, os.stat(path).st_mode | stat.S_IEXEC)
+        return path
 
-            os.chmod(path, os.stat(path).st_mode | stat.S_IEXEC)
-        return [path]
+    @property
+    def _jinja_executor_template(self):
+        """:return Jinja template of the shell-script executor"""
+        template = self.campaign.process.executor_template
+        if template.startswith('#!'):
+            return ENV.from_string(template)
+        else:
+            return ENV.get_template(template)
+
+    def _write_executor_script(self, ostr):
+        """Write shell script in charge of executing the command"""
+        properties = dict(
+            command=self.command,
+            cwd=os.getcwd(),
+            environment=dict(
+                (var, six.moves.shlex_quote(value))
+                for var, value in
+                (self.execution.get('environment') or {}).items()
+            ),
+        )
+        self._jinja_executor_template.stream(**properties).dump(ostr)
 
     @cached_property
     def command(self):
-        """get command to execute
-
-        :return: list of string
-        """
+        """:return command to execute inside the generated shell-script"""
         benchmark_config = self.parent.parent.parent.config
-        if self.execution.get('shell', False):
-            exec_prefix = benchmark_config.get('exec_prefix') or ""
-            if exec_prefix:
-                self.execution['command'][-1].insert(0, exec_prefix)
-            return self._wrap_in_bash_script(self.execution['command'])
-
         exec_prefix = benchmark_config.get('exec_prefix') or []
-        if not isinstance(exec_prefix, list):
-            exec_prefix = shlex.split(exec_prefix)
-        if not isinstance(self.execution['command'], list):
-            command = shlex.split(self.execution['command'])
+        shell = self.execution.get('shell', False)
+        command = self.execution['command']
+        if isinstance(command, list):
+            command = [
+                arg.format(**self.command_expansion_vars)
+                for arg in command
+            ]
         else:
-            command = self.execution['command']
-        command = list(exec_prefix) + command
-        return [
-            arg.format(**self.command_expansion_vars)
-            for arg in command
-        ]
+            command = command.format(**self.command_expansion_vars)
+        if not shell:
+            if not isinstance(exec_prefix, list):
+                exec_prefix = shlex.split(exec_prefix)
+            if not isinstance(command, list):
+                eax = exec_prefix + shlex.split(command)
+                print(eax)
```

Comment:
that's a leftover from debugging?

Reply:
Oh yes. Thank you.
I have just fixed and repushed.